### ISSUE TITLE
Best way to display a stateless component on the page 

### ISSUE THREAD

**Issue by michalpanek** [toddmotto/angular-styleguide#113]
I've been wondering about how to manage "switch" beetwen dumb/stateless component on the page. 

Suppose we have statefull route component _users_ . _users_ rendering in a template three stateless component, but only one at the same time. How is the best way to manage this ? I thought about use ui-router, and switch components navigating beetwen ui-router state. But I don't think is a good idea.

Is a good way to manage this by _ng-switch_, or _ng-if_,   "switching" beetwen dumb/stateless component, depending on the user interactions and choises ? Or is another, better way ? 

```javascript
<--statefull component template-->
<div ng-switch="$ctrl.switchFlag">
     <one-component ng-switch-when="one"></one-component>
    <two-component ng-switch-when="two"></two-component>
</div>
```

**Comment by siddharth-pandey** [comment]
Based on the example that is given in style guide, there are 3 components, a parent (consumes services) and 2 child (list, details) dumb components that are on the same ui-router state. **How should we manage the cases where :** two different states are required to display a list on one state and details on another state but still uses the parent component so that only this component consumes services?

As if I use ui-router to define two states for above purpose, the `$onChanges` method is not triggered when any of the input bindings change that were resolved using `$resolve` service of ui-router. But it works fine if the child component is inside parent template and uses `$ctrl` instead of `$resolve`. Based on [this SO answer](http://stackoverflow.com/a/38646248/1211329), it makes sense that the watcher only watches the members linked with `$ctrl` and not `$resolve` as ui-router adds extra HTML for this process.

So, what will be a good way to use todo, todoForm on one state (for new item or to display an existing item) and todo, todoList on another state? Todo as a case is quite simple, but you can relate this to more complex types where we need list, details/editing views which may need different states like '/products' for list and '/products/:productId' for details/editing. But still use a parent smart component that consumes service and give data to child dumb components.

**Comment by michalpanek** [comment]
I can said that the only way I achieved this is with sending not single bindings, but sending a one object with bindings inside it. Bindings works also with '&'. Bindings in child component are updates when are changed in parent component, and is possible to fire event callback from child dumb component to parent statefull component. But is still one problem with that approach: _$onChanges_ is doesn't fired when bindings is changing.  This looks like that:

In ui-route conf:
```javascript
$stateProvider
    .state('child', {
      url: "/main",
      component: 'childComponent',
      params: {
        bindingsObj: undefined
      },
      resolve: {
        bindingsObj: ($stateParams) => $stateParams.bindingsObj
      }
    })
```

in parent Smart Component: 
```javascript
vm.bindingsObj = {
      prop: vm.propsToSendToChild,
      onEvent: vm.eventToFire
    }
    
    vm.goToChildComponent = function() {
      $state.go('child', { bindingsObj: vm.bindingsObj })
    }
```

and in child stateless/dumb component is possible to fire _onEvent_ function which fire _eventToFire_ function from smart parent component. And when I change _propsToSendToChild_ in _bindingsObj_ in parent component is also changed in child component. 

I create already plnkr. Added some changes to show what I meant. Is written in es5 to simplification.

**Comment by michalpanek** [comment]
I wrote about my another way to solved that problem in ui-router github, which I founded, but I don't now if it is very good.
